src/libmongoc/tests/test-mongoc-ts-pool.c

```diff
@@ -0,0 +1,109 @@
+#include "mongoc/mongoc-ts-pool-private.h"
+
+#include "TestSuite.h"
+#include "test-libmongoc.h"
+
+static void
+test_ts_pool_empty (void *unused)
+{
+   mongoc_ts_pool *pool = mongoc_ts_pool_new (
+      (mongoc_ts_pool_params){.element_size = sizeof (int)});
+   BSON_ASSERT (mongoc_ts_pool_is_empty (pool));
+   mongoc_ts_pool_free (pool);
+}
+
+
+static void
+test_ts_pool_simple (void *unused)
+{
+   mongoc_ts_pool *pool = mongoc_ts_pool_new (
+      (mongoc_ts_pool_params){.element_size = sizeof (int)});
+   int *item;
+   int *item2;
+
+   item = mongoc_ts_pool_get_existing (pool);
+   BSON_ASSERT (!item);
+
+   item = mongoc_ts_pool_get (pool, NULL);
+   BSON_ASSERT (item);
+   ASSERT_CMPINT (*item, ==, 0);
+   *item = 42;
+   ASSERT_CMPSIZE_T (mongoc_ts_pool_size (pool), ==, 0);
+   mongoc_ts_pool_return (item);
+   ASSERT_CMPSIZE_T (mongoc_ts_pool_size (pool), ==, 1);
+
+   item2 = mongoc_ts_pool_get_existing (pool);
+   BSON_ASSERT (item2);
+   ASSERT_CMPINT (*item2, ==, 42);
+   ASSERT_CMPSIZE_T (mongoc_ts_pool_size (pool), ==, 0);
+
+   mongoc_ts_pool_drop (item2);
+   ASSERT_CMPSIZE_T (mongoc_ts_pool_size (pool), ==, 0);
+
+   mongoc_ts_pool_free (pool);
+}
+
+static int
+_is_int_42 (int *v, void *unused)
+{
+   return *v == 42;
+}
+
+static void
+_set_int_to_7 (int *v, void *unused, bson_error_t *unused2)
+{
+   *v = 7;
+}
+
+/* Declare a pool that contains `int`, sets each new int to seven, and drops
+ * integers that are equal to 42. */
+MONGOC_DECL_SPECIAL_TS_POOL (
+   int, int_pool, void, _set_int_to_7, NULL, _is_int_42)
```

Comment:
Compiling locally with maintainer flags produces warnings for unused functions:

```
/Users/kevin.albertson/code/mongo-c-driver/src/libmongoc/tests/test-mongoc-ts-pool.c:60:1: warning: unused function 'int_pool_clear' [-Wunused-function]
MONGOC_DECL_SPECIAL_TS_POOL (
^
/Users/kevin.albertson/code/mongo-c-driver/src/libmongoc/src/mongoc/mongoc-ts-pool-private.h:286:28: note: expanded from macro 'MONGOC_DECL_SPECIAL_TS_POOL'
   static BSON_INLINE void PoolName##_clear (PoolName p)                      \
                           ^
<scratch space>:41:1: note: expanded from here
int_pool_clear
^
/Users/kevin.albertson/code/mongo-c-driver/src/libmongoc/tests/test-mongoc-ts-pool.c:60:1: warning: unused function 'int_pool_drop' [-Wunused-function]
/Users/kevin.albertson/code/mongo-c-driver/src/libmongoc/src/mongoc/mongoc-ts-pool-private.h:307:28: note: expanded from macro 'MONGOC_DECL_SPECIAL_TS_POOL'
   static BSON_INLINE void PoolName##_drop (ElementType *elem)                \
                           ^
<scratch space>:45:1: note: expanded from here
int_pool_drop
^
/Users/kevin.albertson/code/mongo-c-driver/src/libmongoc/tests/test-mongoc-ts-pool.c:60:1: warning: unused function 'int_pool_is_empty' [-Wunused-function]
/Users/kevin.albertson/code/mongo-c-driver/src/libmongoc/src/mongoc/mongoc-ts-pool-private.h:317:27: note: expanded from macro 'MONGOC_DECL_SPECIAL_TS_POOL'
   static BSON_INLINE int PoolName##_is_empty (PoolName p)                    \
                          ^
<scratch space>:47:1: note: expanded from here
int_pool_is_empty
^
/Users/kevin.albertson/code/mongo-c-driver/src/libmongoc/tests/test-mongoc-ts-pool.c:60:1: warning: unused function 'int_pool_visit_each' [-Wunused-function]
/Users/kevin.albertson/code/mongo-c-driver/src/libmongoc/src/mongoc/mongoc-ts-pool-private.h:322:28: note: expanded from macro 'MONGOC_DECL_SPECIAL_TS_POOL'
   static BSON_INLINE void PoolName##_visit_each (                            \
                           ^
<scratch space>:48:1: note: expanded from here
int_pool_visit_each
^
4 warnings generated.
```

Can you add casts for each function, like `(void) int_pool_clear;` to resolve these warnings?

Reply:
I believed that adding `BSON_INLINE` to these functions would erase these warnings. I'll have to dig deeper.

Reply:
Should now be fixed.